### PR TITLE
🔨 Switch PaymentInfo to use useOvermind

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Follow-up of #2763

Things I did extra:
- Change `paymentDetailsUpdated`'s signature to accept a `string` instead of `{ token: string }`, because it only has 1 argument
- Type `updatePaymentDetails` function
- Extract `Body` into a separate component instead of doing it all in the `render` function (just a tiny bit cleaner imo)

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3064/commits/f7ac7da118d07eba23260bea924cb04a82a04dd2 is at https://pr3064.build.csb.dev/s/new.

File: packages/app/src/app/pages/common/Modals/PreferencesModal/PaymentInfo/index.tsx [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
+    actions: {
+      preferences: { paymentDetailsRequested, paymentDetailsUpdated },
+    },
     state: {
       preferences: {
-        paymentDetailError,
-        paymentDetails: paymentDetailsFromHooks,
         isLoadingPaymentDetails,
+        paymentDetailError,
+        paymentDetails: { brand, last4, name },
```
Comment (SaraVieira) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/PaymentInfo/index.tsx]: descontring them here will give an error on opening of cannot read brand of undefined

**Comment by MichaelDeBoey** [comment]
@SaraVieira Will take a look at this one later tonight 🙂

**Comment by SaraVieira** [comment]
I am gonna take a look at this as it would be awesome to have it :)

**Pull request merged**